.github/copilot-instructions.md

```diff
@@ -0,0 +1,254 @@
+- **AISKULight/**: Lightweight version of the SDK
+- **shared/**: Core shared libraries (AppInsightsCore, AppInsightsCommon, 1ds-core-js)
+- **extensions/**: Plugin-based extensions (analytics, dependencies, React, Angular, etc.)
+- **channels/**: Data transmission channels (online, offline, tee)
+- **tools/**: Build and development tools
+- **examples/**: Sample implementations
+
+### Key Technologies
+- **TypeScript/JavaScript**: Primary languages (ES5 target for browser compatibility)
+- **Rush**: Monorepo management tool
+- **Rollup**: Module bundler
+- **Grunt**: Task runner
+- **Dynamic Proto**: Dynamic prototype pattern for performance
+
+## Code Style & Patterns
+
+### TypeScript/JavaScript Conventions
+- Use **ES5-compatible** syntax for browser support
+- Prefer `function` declarations over arrow functions for better IE compatibility
+- Use `var` instead of `let/const` in some contexts for ES5 compatibility
+- Always use semicolons
+- Use 4-space indentation
+- Maximum line length: 140 characters
+
+### Naming Conventions
+- **Classes**: PascalCase (e.g., `PageViewManager`, `TelemetryContext`)
+- **Interfaces**: PascalCase with `I` prefix (e.g., `ITelemetryItem`, `IPageViewTelemetry`)
+- **Methods/Functions**: camelCase (e.g., `trackPageView`, `sendTelemetry`)
+- **Constants**: UPPER_SNAKE_CASE (e.g., `MAX_DURATION_ALLOWED`)
+- **Private variables**: underscore prefix (e.g., `_logger`, `_hasInitialized`)
+
+### Dynamic Proto Pattern
+This project uses a unique `dynamicProto` pattern for performance optimization:
+
+```typescript
+export class MyClass {
+    constructor() {
+        dynamicProto(MyClass, this, (_self, _base) => {
+            // Instance methods and properties defined here
+            _self.myMethod = () => {
+                // Method implementation
+            };
+        });
+    }
+}
+```
+
+### Error Handling
+- Use `_throwInternal` for logging diagnostic errors
+- Always include telemetry context in error messages
+- Use appropriate logging severity levels: `CRITICAL`, `WARNING`, `INFORMATION`
+
+```typescript
+_throwInternal(_logger,
+    eLoggingSeverity.WARNING,
+    _eInternalMessageId.InvalidDurationValue,
+    "Invalid page load duration value",
+    { total, network, request, response, dom });
+```
+
+### Performance Considerations
+- Minimize object allocations in hot paths
+- Use `doPerf()` wrapper for performance tracking
+- Avoid synchronous operations that could block the browser
+- Implement lazy initialization where possible
+- Use object pooling for frequently created objects
+
+## Browser Compatibility
+
+### Target Support
+- **Modern browsers**: Chrome, Firefox, Safari, Edge
+- **Legacy support**: Internet Explorer 8+ (ES5 compatibility required)
+- **Mobile browsers**: iOS Safari, Android Chrome
+
+### Compatibility Patterns
+- Feature detection over browser detection
+- Graceful degradation for missing APIs
+- Polyfills for missing functionality (when necessary)
```

Comment:
Using existing polyfills rather than generating its own new polyfills (hopefully, we don't need to create too many more of these -- although as we move forward we might need additional functions)